inference-engine/src/gna_plugin/gna_plugin.cpp

```diff
@@ -904,10 +914,8 @@ uint32_t GNAPlugin::QueueInference(const InferenceEngine::BlobMap &inputs, Infer
             }
         }
 
-        auto dims = input.second->getTensorDesc().getDims();
-
         auto  importedElements = is2D ? dims[dims.size() - 1] : dims[dims.size() - 1] * dims[dims.size() - 2] * dims[dims.size() - 3];
-        auto  importedFrames = dims[0];
+        auto  importedFrames = is3D ? 1 : dims[0];
         auto  targetGroups = is2D ? dims[dims.size() - 2] : dims[0]; // TODO: no proper support for groups yet
```

Comment:
not sure, should it not be dims[dims.size()-1] for 3d?

Reply:
3D case is CHW - it is a layout for a single image